### PR TITLE
Added command to create a promote pipeline job. Changed the version c…

### PR DESCRIPTION
**Pull request by satr**
…ommand

### CODE REVIEW

File: cmd/buildDeployApplication.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func init() {
 	if buildDeployApplicationEnabled {
 		createJobCmd.AddCommand(buildDeployApplicationCmd)
 		buildDeployApplicationCmd.Flags().StringP("application", "a", "", "Name of the application to build-deploy")
-		buildDeployApplicationCmd.Flags().StringP("branch", "b", "master", "Branch to build-deploy from")
+		buildDeployApplicationCmd.Flags().StringP("branch", "b", "main", "Branch to build-deploy from")
```
Comment (nilsgstrabo) [inline cmd/buildDeployApplication.go:77]: Chaning default to "main" is a breaking change. I think we should keep the old default

File: cmd/deployApplication.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ const deployApplicationEnabled = true
 var deployApplicationCmd = &cobra.Command{
 	Use:   "deploy",
 	Short: "Will trigger deploy of a Radix application",
-	Long:  `Triggers deploy of a Radix application according to the radix config in its repository's master branch.`,
+	Long:  `Triggers deploy of a Radix application according to the radix config in its repository's main branch.`,
```
Comment (nilsgstrabo) [inline cmd/deployApplication.go:31]: Chaning default to "main" is a breaking change. I think we should keep the old default

File: cmd/promoteApplication.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		return nil
+	},
+}
+
+func init() {
+	if promoteApplicationEnabled {
+		createJobCmd.AddCommand(promoteApplicationCmd)
+		promoteApplicationCmd.Flags().StringP("application", "a", "", "Name of the application to be promoted")
+		promoteApplicationCmd.Flags().StringP("deployment", "d", "", "Name of a deployment to be promoted")
+		promoteApplicationCmd.Flags().StringP("from", "", "", "The deployment source environment")
```
Comment (nilsgstrabo) [inline cmd/promoteApplication.go:81]:
```suggestion
		promoteApplicationCmd.Flags().StringP("from-environment", "", "", "The deployment source environment")
```
Make the argument name more explanatory

File: cmd/promoteApplication.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		promoteApplicationCmd.Flags().StringP("to", "", "", "The deployment target environment")
```
Comment (nilsgstrabo) [inline cmd/promoteApplication.go:82]:
```suggestion
		promoteApplicationCmd.Flags().StringP("to-environment", "", "", "The deployment target environment")
```
Make the argument name more explanatory

File: cmd/promoteApplication.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	Short: "Will trigger promote of a Radix application",
+	Long:  `Triggers promote of a Radix application deployment`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		appName, err := getAppNameFromConfigOrFromParameter(cmd, "application")
+		if err != nil {
+			return err
+		}
+
+		deploymentName, _ := cmd.Flags().GetString("deployment")
+		fromEnvironment, _ := cmd.Flags().GetString("from")
```
Comment (nilsgstrabo) [inline cmd/promoteApplication.go:40]:
```suggestion
		fromEnvironment, _ := cmd.Flags().GetString("from-environment")
```
Make the argument name more explanatory

File: cmd/promoteApplication.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+		toEnvironment, _ := cmd.Flags().GetString("to")
```
Comment (nilsgstrabo) [inline cmd/promoteApplication.go:41]:
```suggestion
		toEnvironment, _ := cmd.Flags().GetString("to-environment")
```
Make the argument name more explanatory

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
A few suggestions

File: cmd/buildDeployApplication.go (hunk 1 above)
Comment (satr) [reply cmd/buildDeployApplication.go]: Reverted

File: cmd/deployApplication.go (hunk 2 above)
Comment (satr) [reply cmd/deployApplication.go]: Reverted

**Comment by nilsgstrabo** [comment]
Hi Sergey,

No working when on vacation 😊
Happy new year tomorrow my good colleague and friend 🥳🍾

fre. 30. des. 2022, 23:42 skrev Sergey Smolnikov ***@***.***>:

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**